distro/src/bin/hz

```diff
@@ -20,19 +20,75 @@ PRGDIR=`cd "$PRGDIR"; pwd`
 # set the parent directory as HAZELCAST_HOME.
 export HAZELCAST_HOME=${HAZELCAST_HOME:-$(cd "$PRGDIR/.." >/dev/null; pwd -P)}
 
-if [ $JAVA_HOME ] ; then
-    RUN_JAVA=$JAVA_HOME/bin/java
-else
-    RUN_JAVA=`command -v java 2>/dev/null`
+# common.sh needs it
+SCRIPT_DIR="$PRGDIR"
+
+. "$PRGDIR"/common.sh
+
+
+if [ "${PROMETHEUS_PORT}" ]; then
+    echo "Prometheus enabled on port ${PROMETHEUS_PORT}"
+    PROMETHEUS="-javaagent:${HAZELCAST_HOME}/lib/jmx_prometheus_javaagent-0.14.0.jar=${PROMETHEUS_PORT}:${HAZELCAST_HOME}/config/jmx_agent_config.yaml"
 fi
 
-if [ -z $RUN_JAVA ] ; then
-    echo "Error: JRE not found. Please install JRE 8 or higher in your PATH or set JAVA_HOME appropriately"
-    exit 1
+readJvmOptionsFile "jvm.options"
+
+if [ "$HAZELCAST_LICENSE_KEY" ]; then
+    LICENSING_OPT="-Dhazelcast.enterprise.license.key=${HAZELCAST_LICENSE_KEY}"
 fi
 
-export CLASSPATH="$HAZELCAST_HOME/lib/hazelcast-command-line-CLI_VERSION.jar:$HAZELCAST_HOME/lib/hazelcast-HZ_VERSION.jar:$HAZELCAST_HOME/bin/download:$HAZELCAST_HOME/bin/download/hazelcast-download.properties"
+JAVA_OPTS_ARRAY=(\
+$JDK_OPTS \
+"-Dhazelcast.logging.type=log4j2" \
+"-Dlog4j.configurationFile=file:$HAZELCAST_HOME/config/log4j2.properties" \
+"-Dhz.home=$HAZELCAST_HOME" \
+"-Dhazelcast.config=$HAZELCAST_HOME/config/hazelcast.xml" \
+"-Djet.custom.lib.dir=$HAZELCAST_HOME/custom-lib" \
+$PROMETHEUS \
+$LICENSING_OPT\
+$JVM_OPTIONS \
+$JAVA_OPTS \
+)
 
-HZ_CMD="${RUN_JAVA} -cp ${CLASSPATH} ${JAVA_OPTS} -Dhazelcast.commandline.workingdirectory=${HAZELCAST_HOME} com.hazelcast.commandline.HazelcastCommandLine"
+if [ -z "$LOGGING_LEVEL" ]; then
+    export LOGGING_LEVEL="INFO"
+fi
 
-${HZ_CMD} "$@"
+if [ "$(arch)" == "s390x" ]; then
+  export LOGGING_PATTERN="%d [%highlight{\${LOG_LEVEL_PATTERN:-%5p}}{FATAL=red, ERROR=red, WARN=yellow, INFO=green, DEBUG=magenta}][%style{%t{1.}}{cyan}] [%style{%-10c}{blue}]: %m%n"
+else
+  export LOGGING_PATTERN="%d [%highlight{\${LOG_LEVEL_PATTERN:-%5p}}{FATAL=red, ERROR=red, WARN=yellow, INFO=green, DEBUG=magenta}] [%style{%t{1.}}{cyan}] [%style{%c{1.}}{blue}]: %m%n"
+fi
+
+
+CLASSPATH="$HAZELCAST_HOME/lib:$HAZELCAST_HOME/bin/user-lib:$HAZELCAST_HOME/bin/user-lib/*"
+for filename in "$HAZELCAST_HOME"/lib/*; do
+  case "$filename" in
+    picocli*)
+      ;;
+    *)
+      CLASSPATH="$CLASSPATH:$filename"
+      ;;
+  esac
+done
+
+J_OPTS=""
+for arg in "${JAVA_OPTS_ARRAY[@]}"
+do
+  if [ "$J_OPTS" = "" ]; then
+    J_OPTS="$arg"
```

Comment:
Curious - what does this `J_OPTS` loop do? :)

Reply:
CLI expects format of the `JAVA_OPTS` to have options be joined by a comma, so `J_OPTS` is just reformatted version of `JAVA_OPTS`.

Reply:
Got it, thanks.